File: front_end/dynamic-component/src/app/ias-html-component.ts
import { ComponentType } from './component-type';

export abstract class iasHtmlComponent {
    protected _caption: string;
    protected _x: number;
    protected _y: number;
    protected _z: number;
    protected _componentType: ComponentType;

    get x(): number {
        return this._x;
    }
    set x(theX: number) {
        this._x = theX;
    }
    get y(): number {
        return this._y;
    }

    set y(y: number) {
        this._y = y;
    }
    get z(): number {
        return this._z;
    }

    set z(z: number) {
        this._z = z;
    }
    get caption(): string {
        return this._caption;
    }

    set caption(caption: string) {
        this._caption = caption;
    }
    get componentType(): ComponentType {
        return this._componentType;
    }

    set componentType(componentType: ComponentType) {
        this.componentType = componentType;
    }

}

File: src/main/java/kim/eren/entity/iasHtmlComponent.java
package kim.eren.entity;

public abstract class iasHtmlComponent {

	protected String caption;
	protected int x;
	protected int y;
	protected ComponentType componentType;

	protected iasHtmlComponent() {

	}

	public String getCaption() {
		return caption;
	}

	public void setCaption(String caption) {
		this.caption = caption;
	}

	public int getX() {
		return x;
	}

	public void setX(int x) {
		this.x = x;
	}

	public int getY() {
		return y;
	}

	public void setY(int y) {
		this.y = y;
	}

	public ComponentType getComponentType() {
		return componentType;
	}

	public void setComponentType(ComponentType componentType) {
		this.componentType = componentType;
	}
	
	

}
